### ISSUE TITLE
`indexStatus` returns too many indexes in 2.1

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4868]
I have a table `t1` with two indexes `a` and `b`.
```js
r.table('t1').indexStatus('a')
[
    {
        "function": <binary, 100 bytes, "24 72 65 71 6c 5f..."> ,
        "geo": false ,
        "index": "a" ,
        "multi": false ,
        "outdated": false ,
        "ready": true
    } ,
    {
        "function": <binary, 100 bytes, "24 72 65 71 6c 5f..."> ,
        "geo": false ,
        "index": "b" ,
        "multi": false ,
        "outdated": false ,
        "ready": true
    }
]
```

```js
r.table('t1').indexStatus('b')
[
    {
        "function": <binary, 100 bytes, "24 72 65 71 6c 5f..."> ,
        "geo": false ,
        "index": "b" ,
        "multi": false ,
        "outdated": false ,
        "ready": true
    }
]
```

As you can see when I query for the index status for index `"a"`, `indexStatus` also returns the status of `"b"`. When I query for `"b"`, it works as expected.

This used to work in 2.0.x

**Comment by danielmewes** [comment]
A fix is up in review 3243

**Comment by danielmewes** [comment]
Fixed in `next` b98848f and `v2.1.x` 5360eb5
